src/main/java/oracle/r2dbc/impl/UserDefinedType.java

```diff
@@ -0,0 +1,23 @@
+package oracle.r2dbc.impl;
+
+import io.r2dbc.spi.Type;
+
+public final class UserDefinedType implements Type {
```

Comment:
We must write a JavaDoc before merging any code. The doc should capture the rationale which was present in the mind of the programmer who wrote the code: What problem was the code intended to solve, and how did the programmer intend for their code to be used?